metinhancezik/healthy-life-assistant
# # Authentication Implementation

## Changes
- Implemented user registration with AWS Cognito
- Implemented login functionality
- Added JWT token handling
- Created protected routes
- Added auth context for global state management

## Setup Requirements
1. Copy `.env.example` to `.env`
2. Fill in the following environment variables:
   - DATABASE_URL
   - COGNITO_REGION
   - COGNITO_USER_POOL_ID
   - COGNITO_CLIENT_ID

## Testing
1. Register a new user
2. Verify email
3. Login with credentials
4. Test protected routes

## Notes
- AWS Cognito credentials are required
- PostgreSQL database should be running   
- Login is still doesnot work, only register works
